File: 09. 回文数.py
class Solution:

    def isPalindrome(self, x: int) -> bool:
        if (x < 0):
            return False
        a = x
        y = 0
        while (a > 0):
            y = y * 10 + a % 10
            a = a // 10
        return (x == y)


s = Solution()
if s.isPalindrome(121):
    print("yes")
else:
    print("no")

File: 09. 回文数.cpp
#include <iostream>
#include <string>
#include <unordered_set>
using namespace std;

class Solution
{
public:
    bool isPalindrome(int x)
    {
        int a = x;
        if (a < 0)
            return false;
        long int y = 0;
        while (a > 0)
        {
            y = y * 10 + a % 10;
            a = a / 10;
        }
        return (x == y);
    }
};

int main()
{
    Solution s;
    if (s.isPalindrome(121))
    {
        cout << "yes" << endl;
    }
    else
        cout << "no" << endl;
}